src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,58 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASS_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.equals(new User())) {
+            throw new RegistrationException("Fill in the data fields");
+        }
+
+        if (user.getLogin() == null) {
+            throw new RegistrationException(("Login missing"));
+        }
+
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login must be at least "
+                    + MIN_LOGIN_LENGTH + " characters");
+        }
+
+        if (user.getLogin().matches(".*\\s.*")) {
+            throw new RegistrationException("Password must contain no spacing");
+        }
```

Comment:
do we have this requirement?
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/c3f1de9d-223a-4364-89da-1737c796401b)